### ISSUE TITLE
Problem with our run-commands since version 17.2.8: CreateProcessW failed: Falscher Parameter. (os error 87)

### ISSUE THREAD

**Issue by dbk-tbartsch** [nrwl/nx#22517]
### Current Behavior

Beginning with 17.3.0 of nx we are running into a problem. All our targets for all our projects are failing. The task shortly starts and abruptly crahes without any error message. This problem only occurs with windows cmd or powershell, WSL and TeamCity-Agents (running on Linux) as well as git bash are quite fine.

```typescript
    "mock": {
      "command": "prism mock apps/applications/feedback-bff/openapi.yaml"
    }
```

With the latest release of 18.1.3 we are able to see an error message so we deciced to reach out to you, guys. The error message now states:
```
Error: CreateProcessW `"C:\\WINDOWS\\system32\\cmd.exe /C \"spectral lint gen/apps/feedback-bff/openapi.yaml --ruleset spectral/.spectral-external.yaml ${TC_SPECTRAL_FORMAT}\"\0"` in cwd `Some("D:\\home\\openapi-specs\0")` failed: Falscher Parameter. (os error 87)
```

We are a bit confused about the `\0` at the end of the command respectively at the end of the "cwd".

Additionally we made one observation. If we define a target like this:
```typescript
    "mock": {
      "executor": "nx:run-commands",
      "options": {
        "commands": [
          "prism mock apps/applications/feedback-bff/openapi.yaml"
        ]
      }
    }
```
The command is executed normally. Unfortunately this only applies to `nx:run-commands`, other commands (e.g. your own executir from "tools") are not working.

### Expected Behavior

All targets are running as in v 17.2.8.

### Steps to Reproduce

1.

### Nx Report

```shell
NX   Report complete - copy this into the issue template

Node   : 18.13.0
OS     : win32-x64
npm    : 8.19.3

nx             : 18.1.3
@nx/js         : 18.1.3
@nx/jest       : 18.1.3
@nx/linter     : 18.1.3
@nx/eslint     : 18.1.3
@nx/workspace  : 18.1.3
@nx/devkit     : 18.1.3
@nx/plugin     : 18.1.3
@nrwl/tao      : 18.1.3
typescript     : 5.4.3
---------------------------------------
Community plugins:
@trumbitta/nx-plugin-openapi : 1.12.1
---------------------------------------
Local workspace plugins:
         tools
```

### Failure Logs

```shell
NX   ERROR: Something went wrong in run-commands - CreateProcessW `"C:\\WINDOWS\\system32\\cmd.exe /C \"spectral lint gen/apps/feedback-bff/openapi.yaml --ruleset spectral/.spectral-external.yaml ${TC_SPECTRAL_FORMAT}\"\0"` in cwd `Some("D:\\home\\openapi-specs\0")` failed: Falscher Parameter. (os error 87)

Error: ERROR: Something went wrong in run-commands - CreateProcessW `"C:\\WINDOWS\\system32\\cmd.exe /C \"spectral lint gen/apps/feedback-bff/openapi.yaml --ruleset spectral/.spectral-external.yaml ${TC_SPECTRAL_FORMAT}\"\0"` in cwd `Some("D:\\home\\openapi-specs\0")` failed: Falscher Parameter. (os error 87)
    at default_1 (D:\home\openapi-specs\node_modules\nx\src\executors\run-commands\run-commands.impl.js:60:15)
    at process.processTicksAndRejections (node:internal/process/task_queues:95:5)
    at async TaskOrchestrator.applyFromCacheOrRunTask (D:\home\openapi-specs\node_modules\nx\src\tasks-runner\task-orchestrator.js:223:53)
    at async TaskOrchestrator.executeNextBatchOfTasksUsingTaskSchedule (D:\home\openapi-specs\node_modules\nx\src\tasks-runner\task-orchestrator.js:76:13)
    at async Promise.all (index 0)
    at async TaskOrchestrator.run (D:\home\openapi-specs\node_modules\nx\src\tasks-runner\task-orchestrator.js:53:9)
    at async defaultTasksRunner (D:\home\openapi-specs\node_modules\nx\src\tasks-runner\default-tasks-runner.js:18:16)
    at async anyFailuresInPromise (D:\home\openapi-specs\node_modules\nx\src\tasks-runner\run-command.js:227:26)
    at async invokeTasksRunner (D:\home\openapi-specs\node_modules\nx\src\tasks-runner\run-command.js:192:23)
    at async D:\home\openapi-specs\node_modules\nx\src\tasks-runner\run-command.js:89:24

Process finished with exit code 1
```

### Operating System

- [ ] macOS
- [ ] Linux
- [X] Windows
- [ ] Other (Please specify)

### Additional Information

Unfortunately we are not able to provide a guide for reproduction. Maybe this seems to be a company-specific configuration. We created a new monorepo on our machines and all targets are failing as well. When we use our private Windows-machines the targets are running fine.

Do you have an idea what we could try to fix this problem? We are currently stuck on version 17.2.8.

**Comment by gerrG1t** [comment]
We have the same problem

**Comment by AgentEnder** [comment]
Hey, its been a while since this issue has been opened. My bad on not getting to it sooner. Can someone validate that this still occurs on latest? Without repro I am kind of stuck with crawling code and trying to figure it out and for these shell specific things that isn't great.

**Comment by dbk-tbartsch** [comment]
Thanks for comming back to this.

We were able to find a workaround in one of your commits, so that we could update to the latest version. We added a `.env`-file with `NX_NATIVE_COMMAND_RUNNER=false` in it. All targets were able to run without the error.

After your comment we tried to switch it to `true` again and it's now also working like before. We tried this within three of our repositories. Therefore I think this issue can be closed.
